frame/support/procedural/src/lib.rs

```diff
@@ -1738,3 +1738,54 @@ pub fn origin(_: TokenStream, _: TokenStream) -> TokenStream {
 pub fn composite_enum(_: TokenStream, _: TokenStream) -> TokenStream {
 	pallet_macro_stub()
 }
+
+/// An attribute macro that can be attached to a module declaration. Doing so will
+/// make it available for import later.
+#[proc_macro_attribute]
+pub fn pallet_section(attr: TokenStream, tokens: TokenStream) -> TokenStream {
+	match macro_magic::mm_core::export_tokens_internal(attr, tokens, false) {
```

Comment:
we should also ensure this can only be applied to a module
```suggestion
        let tokens_clone = tokens.clone();
        // ensure this can only be attached to a module
        let _mod = parse_macro_input!(tokens_clone as ItemMod);
        
        // use macro_magic's export_tokens as the internal implementation otherwise
	match macro_magic::mm_core::export_tokens_internal(attr, tokens, false) {
```